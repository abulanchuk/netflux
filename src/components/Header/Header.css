.header-container {
    display: flex;
    flex-direction: row;
    justify-content: space-between ;
    align-items: center;
    position: absolute;
    padding-top: 3vw;
    z-index: 2;

}

.header-container__add-movie-button {
    color: #f65261;
    width: 15vw;
    height: 3vw;
    font-weight: bolder;
    background: #ffffff;
    opacity: 0.7;
    border: 0;
    font-size: 2vw;
    margin-left: 55vw;
    align-items: center;
    justify-content: center;
    cursor: pointer;
}

.header-container__add-movie-button:hover{
    opacity: 0.9;
}

.header-container__loop{
    display: flex;
    width: 4vw;
    height: 4vw;
    margin-left: 70vw;
    margin-top: 3vw;
    cursor: pointer;
    background: transparent;
    border: 0;
}

