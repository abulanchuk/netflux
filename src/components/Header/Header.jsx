import React from "react";
import './Header.css';
import Logo from "../logo/Logo";

const Header = ({showButton}) => {
    return <div className="header-container">
        <Logo/>
        {showButton && <button  type="button" className={"header-container__add-movie-button"}>+ADD MOVIE</button>}
    </div>
}

export default Header;
