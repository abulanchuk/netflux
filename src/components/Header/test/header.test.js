import React from "react";
import renderer from "react-test-renderer";
import Header from "../Header";
import Enzyme, {shallow} from 'enzyme';
import Adapter from '@cfaester/enzyme-adapter-react-18';


test("First snapshot test ", () => {
    const component = renderer.create(
        <Header/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

test("show button ADD MOVIE if showButton is true and selected movie is false ", () => {
    const component = renderer.create(
        <Header showButton={true}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})


Enzyme.configure({adapter: new Adapter()});
it('show button ADD MOVIE if showButton is true and selected movie is false', () => {
    const component = shallow(<Header showButton={true}/>);
    expect(component.find(".header-container__add-movie-button").exists()).toBeTruthy();
    expect(component.find(".header-container__loop").exists()).toBeFalsy();
});

it('show loop button if showButton is true and selected movie is true', () => {
    const component = shallow(<Header showButton={true} selectedMovie={true}/>);
    expect(component.find(".header-container__add-movie-button").exists()).toBeFalsy();
    expect(component.find(".header-container__loop").exists()).toBeTruthy();
});

it('should call setSelectedMovie with null on button click', () => {
    const mock = jest.fn();
    const component = shallow(<Header showButton={true} selectedMovie={true} setSelectedMovie={mock}/>);
    const button = component.find(".header-container__loop");
    button.simulate("click");
    expect(mock).toHaveBeenCalledWith(null);
});

it('should show movie modal on click ADD MOVIE button ', () => {
    const component = shallow(<Header showButton={true}/>);
    const button = component.find(".header-container__add-movie-button");
    expect(component.find("AddMovie").exists()).toBeFalsy();
    button.simulate("click");
    expect(component.find("AddMovie").exists()).toBeTruthy();
});



