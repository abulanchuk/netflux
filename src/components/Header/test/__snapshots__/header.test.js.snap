// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First snapshot test  1`] = `
<div
  className="header-container"
>
  <div
    className="logo-container"
  >
    <div>
      <span
        className="logo-container__first-word"
      >
        netflix
      </span>
      <span
        className="logo-container__second-word"
      >
        roulette
      </span>
    </div>
  </div>
</div>
`;

exports[`Second snapshot test  1`] = `
<div
  className="header-container"
>
  <div
    className="logo-container"
  >
    <div>
      <span
        className="logo-container__first-word"
      >
        netflix
      </span>
      <span
        className="logo-container__second-word"
      >
        roulette
      </span>
    </div>
  </div>
  <button
    className="header-container__add-movie-button"
    onClick={[Function]}
    type="button"
  >
    +ADD MOVIE
  </button>
</div>
`;
